### PR TITLE
Updated audit to use local timezone

### PR DESCRIPTION
**Pull request by droberts-ctrlo**
Do I need to update the filters to use `localtime` as well?

### CODE REVIEW

**Comment by abeverley** [comment]
A bit of a rework on this needed I'm afraid... The time should always be stored in the database as UTC, so that it's an absolute value without needing to know the timezone. Then, when it's retrieved, it should be formatted accordingly.

There is already some of this in the code, such as https://github.com/ctrlo/GADS/blob/038b1024bc7e376d7fe0e115aeaf553f4a90155e/lib/GADS/Datum.pm#L283 whereby the timezone is converted to London timezone (automatically adjusting for daylight saving).

What would be better is if there was a central function to convert times accordingly. In the longer-term this could be defined per-user, but in the short-term a simple function such as the following could be added to the `DateTime` class by adding something like the following to `GADS.pm`:

```
Sub::Install::install_sub({
    code => sub {
        my $self = shift;
        # Do not alter original object
        my $clone = $self->clone;
        # Set to baseline timezone if timezone not-existing
        $clone->time_zone->is_floating && $clone->set_time_zone('UTC');
        $clone->set_time_zone('Europe/London'); # Can be customised later
        $clone->strftime('%e %b %Y %H:%M');
    },
    into => "DateTime",
    as   => 'gads_time', # Could be replicated as gads_date for just date
});
```

And then used such as `print $datetime->gads_time`

File: lib/GADS/Audit.pm [hunk 1]
```diff
@@ -167,6 +167,16 @@ sub logs
         schema     => $self->schema,
         init_value => $_->{user},
     )->presentation(type => 'person', site => $site) foreach @logs;
+    
+    my $df = DateTime::Format::Strptime->new(
```
Comment (droberts-ctrlo) [inline lib/GADS/Audit.pm:171]: I don't like this, but it appears the datetime is pulled from the DB as string, so it's the only way I could do it

Comment (abeverley) [reply lib/GADS/Audit.pm:171]: Ah, you're getting bitten by the `$rs->result_class('DBIx::Class::ResultClass::HashRefInflator')` line above. If you remove that they will be returned as `DateTime` objects. You will then need to adjust the other uses of the result objects accordingly (such as the user, I'm not sure how on initial glance)

File: bin/convert-audit.pl
[the remark this answers is not in the document]
Comment (abeverley) [reply bin/convert-audit.pl:1]: Correct :)

File: views/admin/audit.tt [hunk 2]
```diff
@@ -35,7 +35,7 @@
         label = log.id
       },{
         type   = "text"
-        label  = log.user.text
+        label  = log.user.value
```
[the remark this answers is not in the document]
Comment (abeverley) [reply views/admin/audit.tt:38]: I think I'd still like to use the presentation mapping if possible, it seems a shame to remove this functionality. Let's take a look at it.

[the remark this answers is not in the document]
Comment (abeverley) [reply views/admin/audit.tt:38]: Actually, I didn't realise that in the current version the presentation layer is no longer used to render a person with a popover. As such, we can leave this as it is.

**Comment by abeverley** [comment]
I think just one final thing before I merge, I think we should include seconds in the time?

**Pull request merged**